### PR TITLE
Add API client

### PR DESCRIPTION
**Pull request by AntonioJuliano**
Add API functions for placing orders and documentation for them

### CODE REVIEW

File: src/modules/Api.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    fillOrKill: boolean,
+  }): Promise<{ order: ApiOrder }> {
+    const order: LimitOrder = {
+      makerAccountOwner,
+      makerAccountNumber: new BigNumber(makerAccountNumber),
+      makerMarket: new BigNumber(makerMarket),
+      takerMarket: new BigNumber(takerMarket),
+      makerAmount: new BigNumber(makerAmount),
+      takerAmount: new BigNumber(takerAmount),
+      expiration: new BigNumber(Math.round(new Date().getTime() / 1000)).plus(
```
Comment (BrendanChou) [inline src/modules/Api.ts]: expiration=0 doesn't work here (doesn't work as documented)

File: src/types.ts [hunk 2]
```diff
@@ -386,3 +393,71 @@ export interface Operation {
...
+  contractAddress: string;
+  decimals: number;
+  soloMarket: number;
+}
+
+export interface ApiAccount extends ApiModel {
+  owner: string;
+  number: string;
+  balances: {
+    [maketNumber: string]: {
```
Comment (prettymuchbryce) [inline src/types.ts]: `maketNumber` -> `marketNumber`

File: src/modules/Api.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  address,
+  Integer,
+  SigningMethod,
+  ApiOrder,
+  ApiAccount,
+} from '../types';
+import { LimitOrders } from './LimitOrders';
+
+const FOUR_WEEKS_IN_SECONDS = 60 * 60 * 24 * 28;
+const TAKER_ACCOUNT_OWNER = '0xf809e07870dca762B9536d61A4fBEF1a17178092';
```
Comment (prettymuchbryce) [inline src/modules/Api.ts]: nit: ENV variable

File: src/modules/Api.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+const TAKER_ACCOUNT_NUMBER = new BigNumber(0);
+const DEFAULT_API_ENDPOINT = 'https://api.dydx.exchange/';
```
Comment (prettymuchbryce) [inline src/modules/Api.ts]: nit: ENV variable

File: src/modules/Api.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+  Integer,
+  SigningMethod,
+  ApiOrder,
+  ApiAccount,
+} from '../types';
+import { LimitOrders } from './LimitOrders';
+
+const FOUR_WEEKS_IN_SECONDS = 60 * 60 * 24 * 28;
+const TAKER_ACCOUNT_OWNER = '0xf809e07870dca762B9536d61A4fBEF1a17178092';
+const TAKER_ACCOUNT_NUMBER = new BigNumber(0);
```
Comment (prettymuchbryce) [inline src/modules/Api.ts]: nit: ENV variable

File: src/modules/Api.ts (hunk 3 above)
Comment (AntonioJuliano) [reply src/modules/Api.ts]: Solo is not set up to use ENV variables right now because it's a client library

Comment (prettymuchbryce) [reply src/modules/Api.ts]: @AntonioJuliano Well there should probably be some configuration outside of code. An ENV variable would work fine, it's just baked into the code on compile.

**Pull request merged**